lib/ansiblelint/__init__.py

```diff
@@ -129,8 +137,9 @@ def matchyaml(self, file, text):
             if not isinstance(result, list):
                 raise TypeError("{} is not a list".format(result))
 
-            for section, message in result:
-                matches.append(Match(play[ansiblelint.utils.LINE_NUMBER_KEY],
+            for match in result:
```

Comment:
It's probably easier and cleaner to just do it in place:
```suggestion
            for section, *optional_line_num, message in result:
                try:
                    line_number, = optional_line_num
                except ValueError:
                    line_number = play[ansiblelint.utils.LINE_NUMBER_KEY]
```

Reply:
I wanted to do a simple if/else first but flake8 complained about code complexity so I broke it out to a separate function.

Reply:
To me it seems to be better to just check number of returned values instead of try/catching this.

To be honest, I'd much rather return a key:value store so it's easily extendable in the future.

Reply:
EAFP is considered more pythonic: https://stackoverflow.com/a/11360880/595220.

As for flake8, it may be right, I enabled the McCabe complexity check recently because I want to make the project easier to follow and maintain in general.

If it needs to be a separate function, I still think that there's no reason for it to return more than one value: we don't need one-ring-to-rule-them-all type of complexity either.

Reply:
Please try applying this via GitHub UI and see if it'll be okay with the linter.